test/config_test/config_test.cc

```diff
@@ -185,12 +57,21 @@ uint32_t run(const std::string& directory) {
                      }) == unsuported_win32_configs.end()) {
       OptionsImpl options(
           Envoy::Server::createTestOptionsImpl(filename, "", Network::Address::IpVersion::v6));
-      ConfigTest test1(options);
-      envoy::config::bootstrap::v3::Bootstrap bootstrap;
-      Server::InstanceUtil::loadBootstrapConfig(
-          bootstrap, options, ProtobufMessage::getStrictValidationVisitor(), *api);
-      ENVOY_LOG_MISC(info, "testing {} as yaml.", filename);
-      ConfigTest test2(asConfigYaml(options, *api));
+      options.setMode(Server::Mode::Validate);
+      options.setLogLevel(spdlog::level::trace);
+
+      PlatformImpl platform_impl;
+      Event::SimulatedTimeSystem real_time_system;
+      DefaultListenerHooks default_listener_hooks;
+      ProdComponentFactory prod_component_factory;
+
+      try {
+        MainCommonBase base(options, real_time_system, default_listener_hooks,
```

Comment:
I don't think this actually validates the config in `filename`: for the validation to take place `MainCommonBase::run()` needs to be called. Have you tried failing this test intentionally?

Reply:
I added a call to Run locally and the test still passes. I'm working on adding a negative test that attempts to load a bad config to make sure that that fails.

Reply:
a follow up on my comment: `MainCommonBase::run()` doesn't raise an exception on error, but rather returns true if the validation passes. There's also an issue with error reporting, as `MainCommonBase::run()` swallows the exception.